pkg/common/constants.go

```diff
@@ -65,15 +65,41 @@ const (
 	AttachLimitOverrideLabel   = "gke-volume-attach-limit-override"
 )
 
-// doc https://cloud.google.com/compute/docs/disks/hyperdisks#max-total-disks-per-vm
-var Gen4MachineHyperdiskAttachLimitMap = []struct {
+// doc https://cloud.google.com/compute/docs/general-purpose-machines
+// MachineHyperdiskLimit represents the mapping between max vCPUs and hyperdisk (balanced) attach limit
+type MachineHyperdiskLimit struct {
 	max   int64
 	value int64
-}{
+}
+
+// C4 Machine Types - Hyperdisk Balanced Limits
+var C4MachineHyperdiskAttachLimitMap = []MachineHyperdiskLimit{
+	{max: 2, value: 7},
 	{max: 4, value: 15},
-	{max: 8, value: 23},
 	{max: 16, value: 31},
-	{max: 32, value: 49},
-	{max: 64, value: 63},
-	{max: 1024, value: 127},
+	{max: 48, value: 63},
+	{max: 192, value: 127},
```

Comment:
Set this one higher, so that in the future if they have a higher max, we still have some value set. Otherwise we need to backport to earlier versions.

Reply:
The MapNumber function has:
func MapNumber(num int64) int64 {
	for _, r := range Gen4MachineHyperdiskAttachLimitMap {
		if num <= r.max {
			return r.value
		}
	}
	return 0
}

Hmm maybe still we should just return the VolumeLimitSmall

Reply:
The MapNumber has been updated in this PR:

```
func MapNumber(vCPUs int64, limitMap []MachineHyperdiskLimit) int64 {
	for _, limit := range limitMap {
		if vCPUs <= limit.max {
			return limit.value
		}
	}
	return 0
	// Return the last value if vCPUs exceeds all max values
	if len(limitMap) > 0 {
		return limitMap[len(limitMap)-1].value
	}
	return 15
}
```